src/finiteVolume/processes/les.hpp

```diff
@@ -0,0 +1,105 @@
+#ifndef ABLATELIBRARY_LES_H
+#define ABLATELIBRARY_LES_H
+
+#include "eos/transport/transportModel.hpp"
+#include "finiteVolume/fluxCalculator/fluxCalculator.hpp"
+#include "flowProcess.hpp"
+
+namespace ablate::finiteVolume::processes {
+
+class LES : public FlowProcess {
+   private:
+    const std::shared_ptr<eos::EOS> eos;
+    const std::shared_ptr<eos::transport::TransportModel> transportModel;
+
+    // constant values
+    inline const static PetscReal c_k = 0.094;
+    inline const static PetscReal c_e = 1.048;
+    inline const static PetscReal c_p = 1.040;
+
+    /* store turbulent diffusion  data */
+    struct DiffusionData {
+        eos::ThermodynamicTemperatureFunction diffFunction;
+        eos::ThermodynamicFunction computeTemperatureFunction;
+
+        PetscInt numberSpecies;
+        PetscInt numberEV;
+    };
+    DiffusionData diffusionData;
+
+   public:
+    explicit LES(std::shared_ptr<eos::EOS> eos, std::shared_ptr<eos::transport::TransportModel> transportModel = {});
+
+    /**
+     * public function to link this process with the flow
+     * @param flow
+     */
+    void Setup(ablate::finiteVolume::FiniteVolumeSolver& flow) override;
+
+   public:
+    /**
+     * This computes the momentum transfer for SGS model for rhoU
+     * f = "euler"
+     * u = {"euler", "densityYi"}
+     * a = {"yi"}
+     * ctx = lesDiffusionData
+     * @return
+     */
+    static PetscErrorCode lesMomentumFlux(PetscInt dim, const PetscFVFaceGeom* fg, const PetscInt uOff[], const PetscInt uOff_x[], const PetscScalar field[], const PetscScalar grad[],
```

Comment:
Would you mind starting functions with a capital letter? Take a look at https://google.github.io/styleguide/cppguide.html#General_Naming_Rules for guidance.